pvp/controller/control_module.py

```diff
@@ -259,9 +280,15 @@ def __analyze_last_waveform(self):
                 self._DATA_PEEP = np.percentile(pressure[ pressure < mean_pressure], 20 )
                 self._DATA_PIP_PLATEAU  = np.percentile(pressure[ pressure > mean_pressure], 80 )
                 self._DATA_PIP  = np.percentile(pressure[ pressure > mean_pressure], 95 )             #PIP is defined as the maximum, here 95% to account for outliers
-                self._DATA_PIP_TIME = phase[np.min(np.where(pressure > self._DATA_PIP_PLATEAU*0.9))]
-                self._DATA_PEEP_TIME = phase[np.min(np.where(pressure < self._DATA_PEEP))]
-                self._DATA_I_PHASE = phase[np.max(np.where(pressure > self._DATA_PIP_PLATEAU*0.9))]
+                
+                #self._DATA_PIP_TIME = phase[np.min(np.where(pressure > self._DATA_PIP_PLATEAU*0.9))]
+                self._DATA_PIP_TIME = self.__comptest(phase, pressure > self._DATA_PIP_PLATEAU*0.9, 'first')
```

Comment:
this could crash if pressure>self._DATA_PIP was empty (e.g. if all data is 0). The new code makes sure that that doesn't happen

Reply:
It does not affect functionality under normal circumstances